src/other/broadcasting.jl

```diff
@@ -0,0 +1,63 @@
+struct LazyNewColDataFrame
+    df::DataFrame
+    col::Symbol
+end
+
+# we allow LazyNewColDataFrame only for data frames with some columns
+Base.axes(x::LazyNewColDataFrame) = axes(x.df[1])
+
+Base.Broadcast.broadcastable(df::AbstractDataFrame) = Matrix(df)
+
+Base.maybeview(df::AbstractDataFrame, idxs...) = view(df, idxs...)
+
+function Base.maybeview(df::AbstractDataFrame, idxs)
+    if ncol(df) == 0
+        throw(ArgumentError("Broadcasting into a data frame with no columns is not allowed"))
+    end
+    if idxs isa Symbol
+        if !haskey(df, idxs)
+            if !(df isa DataFrame)
+                throw(ArgumentError("column $idxs not found"))
+            end
+            return LazyNewColDataFrame(df, idxs)
+        end
+    end
+    view(df, idxs)
+end
+
+function Base.copyto!(lazydf::LazyNewColDataFrame, bc)
+    if isempty(lazydf.df)
+        throw(ArgumentError("Broadcasting creation of a column of an empty data frame is not allowed"))
+    end
+    T = mapreduce(i -> typeof(bc[i]), promote_type, eachindex(bc); init=Union{})
+    col = Tables.allocatecolumn(T, nrow(lazydf.df))
+    copyto!(col, bc)
+    lazydf.df[lazydf.col] = col
+end
+
+function Base.copyto!(df::AbstractDataFrame, bc)
```

Comment:
Can we use a more specific type for `bc`? I'm not sure we want to officially support this definition of `copyto!` for any type.

Reply:
This is the approach in Base. If I understand @mbauman comment correctly in some cases broadcasting infrastructure does not return `Broadcasted` object.

Reply:
Are you sure? [The docs](https://docs.julialang.org/en/latest/manual/interfaces/#Working-with-Broadcasted-objects-1) seem to say that we can rely on that.

Reply:
The docs say we have to handle `Broadcasted` but do not explicitly say that this is the only option. Anyway - I have tested several natural options and `Broadcasted` is always passed, so I guess we can restrict the signature and if some special cases are found later we can always handle special cases.